lib/ws-browser.js

```diff
@@ -36,15 +50,53 @@ class Transport extends EventEmitter {
     return this.socket;
   }
 
+  _sendBufferedMessages() {
+    let bufferedAmount = this.socket.bufferedAmount;
+    let toSend = '';
+    while (this._sendBuffer.length !== 0) {
+      const message = this._sendBuffer[0];
+      const maxChunkLength = BUFFER_SIZE_LIMIT - bufferedAmount;
+      if (maxChunkLength < 0) {
+        this.socket.close();
+        throw new Error('WebSocket buffer is overflown');
```

Comment:
But if `this.socket.bufferedAmount > BUFFER_SIZE_LIMIT` and there's no exception, doesn't it mean that the actual limit in a particular browser is higher than the value of that constant and we're just dropping the connection for no reason? Instead, we can increase `BUFFER_SIZE_LIMIT` to the value of `this.socket.bufferedAmount` and try again at the next timer tick.

Reply:
I mean, that's the only way for the condition to hit.

Reply:
@aqrln when using only function `_sendBufferedMessages` to send anything over the WebSocket it is not possible to reach the state where this condition will be met. The only way to get this error is to use the WebSocket socket directly and call the `send()` method by accessing the `socket` field of the instance of this W3C WebSocket `Transport` class.
The reasoning behind this condition was explained before in [this thread](https://github.com/metarhia/jstp/pull/381#discussion_r217608269).